proposal/zivzone/proposal.rst

```diff
@@ -0,0 +1,131 @@
+
+
+System architecture
+===================
+The whole library contains different functions in several categories listed below:
+- Activate Fuction
+	- These functions provide the common activate functions for build the network, like Relu and Sigmoid
+- Layers
+	- These functions are used to build the layer in the network, like fully connected layer.  
+- Loss Fuction
+	- We define several common loss function here for evaluate the accuracy of the network, like MSE and L1 Loss.
+- Optimization
+	- We provide several common optimizer for training the network, like SGD.
+- Method
+	- The methods contain the function you can load dataset and autograd for calculate gradient. 
+- Data processing
+	- For data preprocessing
+- Deploy
+	- For deploy the network on different platform.
+- Format/Storage/Data Utilize
+	- Data format transform.
+- Model
+	- Several built-in networks for testing.
+- Case/Application
+	- Some application for example.
+- Visualize
+	- Visualize the result of training and testing.
+- Analyze
+	- Analyze the network like running time, space usage, and flpos counting.
+- Optimize
+	- Provide several method for optimize the model.
+	
+For the user, you may need to define and build the network by using **Layers** and **Activate Fuction**, then loading the dataset and do the data preprocessing by **Method** and **Data processing**.
+Then define your loss function by **Loss Fuction** and **Optimization** for training your network.
+Final, use **Visualize** visualize your training result.
+Additionaly, you can use **Analyze** for analizing your network and use **Optimize** for optimizing your network.
+
+
+API description
+===============
+
+Show how a user programmatically uses your system.  You are supposed to
+implement the system using both C++ and Python.  Describe how a user writes a
+script in the system.  If you provide API in both C++ and Python, describe
+both.
+
+For example, if the user want to build a network (a simple CNN model for classification), they can take the code below as reference:
+```
+#Import function
+from Layers.Convolution2d import Conv2d
+
+# define covolution layer
+# Conv2d(
+	input_channel_number,  # the channel number of input
+	output_channel_number, # the channel number of output
+	kernel_size,           # the size of kernel
+	stride,                # the size of stride
+	padding,               # the size of padding
+	bias                   # True or False for bias
+		)
+layer = Conv2d(3,16,kernel_size = (5,5), stride = (1,1), padding = (2,2),bias = True)
+```
+
+```
+#Import the function for loading data
+from Method.Data import Load_file
+# Load_file('dataset_name','setting')
+# Return the dataset
+Load_file('cifar10', 'train')
+
+```
+
+Engineering infrastructure
+==========================
+Ubuntu 18.04 and Python 3.6 for development and testing.
+When I start to develp one function, i will open a new branch. And I will merge it with master after testing.
```

Comment:
How do you implement the low-level matrix operations code?  numpy is too slow for it.